src/helpers/text_helper.ts

```diff
@@ -224,6 +228,90 @@ export function getContextFontSize(font: string): Pixel {
   return Number(font.match(pxRegex)?.[1]);
 }
 
+export function fittingString(ctx, str: string, maxWidth: number) {
+  var width = ctx.measureText(str).width;
+  var ellipsis = "…";
+  var ellipsisWidth = ctx.measureText(ellipsis).width;
+  if (width <= maxWidth || width <= ellipsisWidth) {
+    return str;
+  } else {
+    var len = str.length;
```

Comment:
`var` :eyes: smells like something from stack overflow :smile: 
If it is, it's good not to re-invent the wheel and use what others have already done, but it requires the reference to the source to credit the original author :wink:

Reply:
Yes, it has been copied from SO :D I added the link to the answer :)